pkg/virt-handler/vm.go

```diff
@@ -605,6 +605,14 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 				if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
 					newInterface.InterfaceName = interfaceStatus.InterfaceName
+
+					// Qemu guest reports IP addresses in x.x.x.x/mask format
+					// remove the mask from the ip address
+					interfaceStatus.Ip = strings.Split(interfaceStatus.Ip, "/")[0]
```

Comment:
Would `net.ParseCIDR("192.0.2.1/24")` help?

Reply:
also, can we get a unit test so we don't goof this up in the future somehow?

Reply:
@rmohr yes that's better. @davidvossel sure I will add unit tests as well.

Reply:
Since we don't own the guest-agent, maybe we should also be cautious here and check if it is with or without subnet provided.


To illustrate the flow:
```
if err := net.ParseCIDR(); err != nil { # with netmask
   if err := net.ParseIP(); err != nil { # without netmask
     log.Error("qemu-guest-agent sent garbage")
     continue # not failing, but rather skipping this ip.
  }
}

```

Reply:
I see that `ParseCIDR` checks if IP address is correct https://golang.org/src/net/ip.go?s=16533:16577#L702 so I think it's not needed to run `ParseIP` again.

I updated tests, let's see if everything passes.

Reply:
It may check if the IP is correct, but it fails if no CIDR is included: https://play.golang.org/p/sSkJYZ3JFDR

Reply:
yeah I realized that when I tested things :) however it's solved as we spoke here https://github.com/kubevirt/kubevirt/pull/4217#discussion_r496522696